lib/ood_core/job/adapters/kubernetes/helper.rb

```diff
@@ -38,18 +38,21 @@ def info_from_json(pod_json: nil, service_json: nil, secret_json: nil, ns_prefix
   #
   # @param container [#to_h]
   #   the input container hash
+  # @param default_env [#to_h]
+  #   Default env to merge with defined env
   # @return  [OodCore::Job::Adapters::Kubernetes::Resources::Container]
-  def container_from_native(container)
+  def container_from_native(container, default_env)
     OodCore::Job::Adapters::Kubernetes::Resources::Container.new(
       container[:name],
       container[:image],
       command: parse_command(container[:command]),
       port: container[:port],
-      env: container.fetch(:env, []),
+      env: container.fetch(:env, {}),
       memory: container[:memory],
       cpu: container[:cpu],
       working_dir: container[:working_dir],
-      restart_policy: container[:restart_policy]
+      restart_policy: container[:restart_policy],
+      default_env: default_env
```

Comment:
`deep_merge` is already available to the helper. Seems like we shouldn't have a `default_env` member variable and instead just merge when creating `@env`. That way we can get rid of the `env` method on the resource too.

https://github.com/OSC/ood_core/blob/087d28596c85a6ec77000a1e2045ba6e218acb63/lib/ood_core/job/adapters/kubernetes/helper.rb#L7

Reply:
I updated to use `merge` since these are not going to be deep hashes, only key/value pairs.